Packs/URLHaus/Integrations/URLHaus/URLHaus.py

```diff
@@ -86,330 +85,624 @@ def test_module(**kwargs):
+        indicator_type=DBotScoreType.URL,
+        reliability=params.get('reliability'),
+        score=Common.DBotScore.NONE
+    )
+    url_indicator = Common.URL(url=uri, dbot_score=dbot_score)
+    human_readable = f'## URLhaus reputation for {uri}\n' \
+                     f'No results!'
+    return CommandResults(
+        readable_output=human_readable,
+        raw_response=url_information,
+        indicator=url_indicator
+    )
+
+
+def process_query_info(url_information: dict, uri: str, params: dict) -> CommandResults:
+    """
+         Process the response.
+
+         Args:
+            url_information (dict): The data retrieved from URLHaus db.
+            uri (str): The queried URL.
+            params (dict): The integration params.
+
+         Returns:
+             result (CommandResults): The CommandResults object representing the url command results.
+    """
+    if url_information['query_status'] == 'ok':
+        return build_context_url_ok_status(url_information, uri, params)
+
+    elif url_information['query_status'] == 'no_results':
+        return build_context_url_no_results_status(url_information, uri, params)
 
+    elif url_information['query_status'] == 'invalid_url':
+        human_readable = f'## URLhaus reputation for {uri}\n' \
+                         f'Invalid URL!'
+        return CommandResults(
+            readable_output=human_readable,
+            raw_response=url_information,
+        )
+    else:
+        raise DemistoException(f'Query results = {url_information["query_status"]}', res=url_information)
+
+
+def run_url_command(url: str, params: dict) -> CommandResults:
+    """
+         Query the url_information from URLHaus db.
+
+         Args:
+            params (dict): The integration params.
+            url (str): The queried URL.
+
+         Returns:
+             result (CommandResults): The CommandResults object representing the url command resultsgit .
+    """
     try:
-        domain_information = query_host_information(domain, kwargs.get('api_url'), kwargs.get('use_ssl')).json()
-
-        ec = {
-            'Domain': {
-                'Name': domain
-            },
-            'DBotScore': {
-                'Type': 'domain',
-                'Vendor': 'URLhaus',
-                'Indicator': domain,
-                'Reliability': kwargs.get('reliability')
-            }
+        url_information = query_url_information(url, params.get('api_url'), params.get('use_ssl')).json()
+    except UnicodeEncodeError:
+        return CommandResults(
+            readable_output='Service Does not support special characters.',
+        )
+    return process_query_info(url_information, url, params)
+
+
+def url_command(params: dict):
+    """
+        Split urls and call run_url_command on each of them.
+
+         Args:
+            params (dict): The integration params.
+    """
+    urls = demisto.args().get('url', '')
+    for url in urls.split(','):
```

Comment:
```suggestion
    for url in argToList(urls):
```
url arg might already be a list given to the argument from the context:
`!url url=${Path.to.URLs}`
argToList() handles this. please change in other commands too.